packages/renderer/src/lib/ui/Typeahead.svelte

```diff
@@ -141,35 +145,55 @@ function makeVisible(): void {
 }
 
 function processInput(): void {
+  searchResults = [];
   loading = true;
-  searchFunction(value)
-    .then(result => {
-      // if the component has been disabled in the meantime
-      if (disabled) {
-        return;
-      }
-      items = result.toSorted((a: string, b: string) => {
-        const dockerIoValue = `docker.io/${value}`;
-        const aStartsWithValue = a.startsWith(value) || a.startsWith(dockerIoValue);
-        const bStartsWithValue = b.startsWith(value) || b.startsWith(dockerIoValue);
-        if ((aStartsWithValue && bStartsWithValue) || (!aStartsWithValue && !bStartsWithValue)) {
-          return a.localeCompare(b);
-        } else if (aStartsWithValue && !bStartsWithValue) {
-          return -1;
-        } else {
-          return 1;
+  for (const [index, searchFunction] of searchFunctions.entries()) {
+    searchFunction(value)
+      .then(result => {
+        // if the component has been disabled in the meantime
+        if (disabled) {
+          return;
+        }
+        searchResults[index] = result.toSorted((a: string, b: string) => {
+          const dockerIoValue = `docker.io/${value}`;
```

Comment:
+1 for https://github.com/podman-desktop/podman-desktop/pull/10573/files#r1908966973 as we should not have pullImage specific code in the Typeahead component